src/com/cafeto/training/AccessClass.java

```diff
@@ -0,0 +1,30 @@
+package tests;
+
+import com.acces.IDataAccess;
+import com.singleton.lazySingleton;
+
+import org.mockito.Mockito;
+
+/**
+ * Created by agomez on 7/6/2015.
+ */
+public class AccesClass
+{
+    lazySingleton instance;
```

Comment:
this shouldn't exists, you should have an API that allow the user to pass its own IDataAccess, and set it to the Singleton, you are not mocking the containing factory, you are mocking the internal singleton, in this case, the IDataAcess.